features/integration/simulate.feature

```diff
@@ -182,3 +182,25 @@ Feature: Simulating transactions
     # Final step to check extra budgets in simulation result
     Then I check the simulation result has power packs extra-opcode-budget with extra budget 2000.
     And the simulation should succeed without any failure message
+
+  @simulate.exec_trace_with_stack_scratch
+  Scenario: Simulate app with response containing stack and scratch changes
+  Given a new AtomicTransactionComposer
+    When I build an application transaction with the transient account, the current application, suggested params, operation "create", approval-program "programs/stack-scratch.teal", clear-program "programs/eight.teal", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes ""
+    And I sign and submit the transaction, saving the txid. If there is an error it is "".
+    And I wait for the transaction to be confirmed.
+    Given I remember the new application ID.
+
+    Given I add the nonce "simulate-with-exec-trace-stack-scratch"
+    When I make a new simulate request.
+    When I create the Method object from method signature "manipulation(uint64)uint64"
+    * I create a new method arguments array.
+    * I append the encoded arguments "AAAAAAAAAAo=" to the method arguments array.
+    * I add a nonced method call with the transient account, the current application, suggested params, on complete "noop", current transaction signer, current method arguments.
+
+    Then I allow exec trace options "stack,scratch" on that simulate request.
+    Then I simulate the transaction group with the simulate request.
+    And the simulation should succeed without any failure message
+
+    Then 4th unit in the "approval" trace at txn-groups path "0" should add to stack "uint64:1", pop from stack by 2, write to "none" scratch slot by "none".
+    Then 31th unit in the "approval" trace at txn-groups path "0" should add to stack "", pop from stack by 1, write to "1" scratch slot by "uint64:18446744073709551615".
```

Comment:
There are no tests with `bytes:`? Since the implementation supports it, I'd recommend having at least one occurrence so we know that code path works.

And the bytes should be encoded in some way that can represent any value, e.g. base64.

Reply:
right, good reminder, I recalled there are bytes appearing in opcode. I can add that in.

Reply:
see [update steps](https://github.com/algorand/algorand-sdk-testing/pull/301/commits/3e14f984882d75b98c8610f6838ce9017fdc2a2b)